assets/js/googlesitekit/widgets/components/WidgetContextRenderer.js

```diff
@@ -35,9 +35,17 @@ const { useSelect } = Data;
 const WidgetContextRenderer = ( props ) => {
 	const { slug, className, Header, Footer } = props;
 
-	const widgetAreas = useSelect( ( select ) => select( STORE_NAME ).getWidgetAreas( slug ) );
+	const widgetAreas = useSelect( ( select ) => {
+		if ( slug ) {
+			return	select( STORE_NAME ).getWidgetAreas( slug );
+		}
+	} );
 	const widgetAreasLength = widgetAreas?.length;
 
+	if ( ! slug ) {
```

Comment:
I was thinking maybe we should use `widgetAreasLength` here instead since this is a bit more clear as to why we're returning early but that would be a change that we didn't have before. `slug` was required before and now it isn't, but a context could also have zero areas returned by the selector for the given slug, in which case we probably shouldn't render it since a context with no areas doesn't make much sense I think. Let's do that – we can use `widgetAreas?.length` inline here and below and do away with the extra variable, although the reference beyond this point wouldn't need the conditional for accessing `length` 🙂 

```suggestion
	if ( ! widgetAreas?.length ) {
```